source/js/components/petition/donation-modal.jsx

```diff
@@ -137,6 +135,19 @@ class DonationModal extends Component {
     );
   }
 
+  generateDonateButtonUrl() {
```

Comment:
Hi @mmmavis!

I updated this function based on the feedback that you left [here](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10892#discussion_r1280940102).

I was wondering if you could take a look and let me know what you think?

It is working well for me, the only question that I have is this:

When a user enters a relative URL (ex: `?form=test`), this method generates a full URL (ex: `http://<url_of_current_page>/?form=test`. This means that a user clicking the button will trigger a page refresh.

Is this OK? 

Thanks!

Reply:
Hey @danielfmiranda , thanks for calling this out. In that case I think we can modify the code so that before `return`, we massage the URL string again.

We can update `const url = new URL(this.props.donateUrl, window.location.href);` to `const url = new URL(this.props.donateUrl);`

And instead of returning `url.toString();`, we can do something like 

```js
const current = new URL(window.location.href);
const baseUrl = current.origin + current.pathname;

return url.toString().replace(baseUrl, "");
```

Reply:
I'm also curious if Python/Django has built-in functions for generating URL like this?  If so, we should probably utilize that.